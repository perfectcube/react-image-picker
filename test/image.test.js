import React from 'react';
import ImagePicker from '../src/react-image-picker';
import renderer from 'react-test-renderer';
import { Map } from 'immutable';

import img1 from '../src/assets/images/kitten/200.jpg'
import img2 from '../src/assets/images/kitten/201.jpg'
import img3 from '../src/assets/images/kitten/202.jpg'
import img4 from '../src/assets/images/kitten/203.jpg'

const imageList = [img1, img2, img3, img4]

test('should accept a default image picked', () => {
    const picked = Map().set(0, img1 + "0");
    const component = renderer.create(
        <ImagePicker
            images={imageList.map((image, i) => ({src: image + i, value: i}))}
            picked={picked}
            />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});