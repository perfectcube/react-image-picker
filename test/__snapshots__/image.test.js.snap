// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept a default image picked 1`] = `
<div
  className="image_picker"
>
  <div
    className="responsive selected"
    onClick={[Function]}
  >
    <img
      className="thumbnail selected"
      src="test-file-stub0"
      style={
        Object {
          "height": 150,
          "objectFit": "cover",
          "width": 150,
        }
      }
    />
    <div
      className="checked"
    >
      <div
        className="icon"
      />
    </div>
  </div>
  <div
    className="responsive"
    onClick={[Function]}
  >
    <img
      className="thumbnail"
      src="test-file-stub1"
      style={
        Object {
          "height": 150,
          "objectFit": "cover",
          "width": 150,
        }
      }
    />
    <div
      className="checked"
    >
      <div
        className="icon"
      />
    </div>
  </div>
  <div
    className="responsive"
    onClick={[Function]}
  >
    <img
      className="thumbnail"
      src="test-file-stub2"
      style={
        Object {
          "height": 150,
          "objectFit": "cover",
          "width": 150,
        }
      }
    />
    <div
      className="checked"
    >
      <div
        className="icon"
      />
    </div>
  </div>
  <div
    className="responsive"
    onClick={[Function]}
  >
    <img
      className="thumbnail"
      src="test-file-stub3"
      style={
        Object {
          "height": 150,
          "objectFit": "cover",
          "width": 150,
        }
      }
    />
    <div
      className="checked"
    >
      <div
        className="icon"
      />
    </div>
  </div>
  <div
    className="clear"
  />
</div>
`;
